### ISSUE TITLE
Pondering out loud

### ISSUE THREAD

**Issue by jdtsmith** [neilgupta/homebridge-analytics#1]
You mention that the only possibility for Homekit logging is creating fake switches to monitor sensor events.  But Homekit certainly does provide live information on all sensors, switches, etc., in the form of updates to the Home App itself!  I'm wondering whether there is any technical limitation that prevents a service based on [HAP-NodeJS](https://github.com/homebridge/HAP-NodeJS) from:

1. Querying for and subscribing to all the get/set messages for all active Homekit accessories on the network.
2. Logging accessory state changes as they occur into a database.

Since you've pondered this already I thought I'd see if there is something basic I'm missing...

**Comment by neilgupta** [comment]
Great question! I actually did investigate that route first. Unfortunately, HAP-NodeJS is an implementation of the HomeKit Accessory Protocol, which is how HomeKit devices talk to the HomeKit database. Unfortunately, HomeKit devices aren't allowed to know what other HomeKit devices are on the network or what they're doing. Only Apple devices can read from the HomeKit database, hence this hacky workaround where we use HomeKit's trigger system to report back to us.

Having said that, there is another more graceful way to solve this problem, but it requires a dedicated AppleTV or iOS device and a sideloaded app. As developers, we _are_ allowed to build HomeKit apps that can read the database. If that app were to run constantly in the foreground, it could report back any changes it sees. I built a proof of concept of this that publishes HomeKit changes to a local MQTT broker here: https://github.com/Hiome/hkbrain-tvos

Hope that helps!
